docs/book/src/developer/architecture/controllers/control-plane.md

```diff
@@ -233,6 +233,18 @@ The `status` object **may** define several fields:
 * `externalManagedControlPlane` - is a bool that should be set to true if the Node objects do not
   exist in the cluster. For example, managed control plane providers for AKS, EKS, GKE, etc, should
   set this to `true`. Leaving the field undefined is equivalent to setting the value to `false`.
+* `paused` - is a condition that reports if the cluster or control plane resource is paused. It should check if 'spec.paused' is set on the cluster, and for the paused annotation on the resource. This is currently optional, but will become required in future.
+```go
+	// Return early and set the paused condition to True if the object or Cluster
+	// is paused.
+	if annotations.IsPaused(cluster, m) {
+		log.Info("Reconciliation is paused for this object, setting Paused condition")
+		conditions.MarkTrue(m, clusterv1.PausedCondition)
```

Comment:
Should this have additional information as to why it's being paused? A reason, a message, a severity?

Reply:
Yeah, this occured to me when working on this. I mean to ask but lost track! 

We don't currently have a helper to set a reason with a `MarkTrue` - I could make one, but was wondering if there was reasoning around not having done so previously? 

e.g: https://github.com/kubernetes-sigs/cluster-api/blob/fe9d5bcc0b9a39f28c7d7c8a7e6e6977644997a0/util/conditions/setter.go#L80-L108

Reply:
Yeah good question, for this condition I definitely imagine it having a reason or message. "PausedAnnotationPresent - Resource is paused as it is annotated with the pause annotation" or "ClusterPaused - Resource is paused due to cluster being paused"

Perhaps we just haven't needed that kind of annotation before?

Reply:
That's an interesting one.
According to https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20200506-conditions.md, condition types MUST have a consistent polarity (i.e. "True = good").

Also
> Condition types in Cluster API core objects MUST represent the operational state of a component in the cluster, where the operational state is when the component is ready to serve application workloads...
> Operations like upgrades, scaling-up, or scaling-down, even if orchestrated in a non-disruptive fashion, MUST be considered as a deviation from the normal operational state of the cluster...

So, if we stick to current conditions standard in Cluster, unfortunately paused isn't a compliant condition, because it doesn't have a positive polarity (and this is why it doesn't fit into existing helpers).

The idea of improving conditions have been around since some time now (see eg https://github.com/kubernetes-sigs/cluster-api/issues/7635 + linked docs); we should figure it out if this goes in the critical path for this change or not

Reply:
> condition types MUST have a consistent polarity (i.e. "True = good").

Well that is an interesting constraint, do you have any idea where that came from? I believe it's pretty common to have conditions that are not `True = good`, for example, Node's have `DiskPressure` and other pressures where `True = bad`

Skimming through the enhancement I have little issue otherwise with what's prescribed there, it seems reasonable apart from this constraint on `True = good` and the allowed suffixes for conditions 🤔 

I don't see any such constraints in the upstream conditions KEP. In fact, in the [API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md) you have:

> Condition type names should make sense for humans; neither positive nor negative polarity can be recommended as a general rule

> So, if we stick to current conditions standard in Cluster, unfortunately paused isn't a compliant condition, because it doesn't have a positive polarity (and this is why it doesn't fit into existing helpers).

In theory we could rename the condition to make it compliant, but, to me at least, that feels like we are cutting our nose to spite our face. Paused is a natural term for what we are trying to represent here, `ReconcileAllowed` or alternatives that fit the convention feel very awkward to me. And as above, this is not recommended in Kube in general.

Also, reviewing [your summary](https://docs.google.com/document/d/1hBQnWWa5d16FOslNhDwYVOhcMjLIul4tMeUgh4maI3w/edit), it appears there are a large number of exceptions to the rules there already? Are those still accurate, I appreciate the doc is 18 months old at this point

Reply:
The assumption about polarity is backed into the condition package everyone is using to aggregate conditions, roll them up across the hierarchy of objects etc. so I don't think we have exceptions as of today.

I also fully agree we should do a new iteration on the package above + conditions, I tried to get this moving a couple of times, happy to pair if someone is interested in it

Reply:
> I also fully agree we should do a new iteration on the package above + overall conditions, happy to pair if someone is interested in it

I'd be happy to take a stab at this, and also to pair on it :) 

Happy to have a quick catch up (meet, slack etc) to clarify expectations as well. Let me know when is good for you!

Reply:
Hm, stupid question. Isn't `Paused` neither good nor bad? I also think it doesn't really make sense to use it in summaries. How would Paused true or false aggregate into e.g. a Ready condition? (in my opinion neither true nor false should influence something like Ready)

For me the concept of polarity (in the sense of good/bad) doesn't make sense for something like "Paused".

(apart from "Paused" being clearly better then something like "Unpaused" because "Paused" doesn't invert the meaning like "Unpaused" does, so it's easier to grasp)

Reply:
(Discussed this a bit with Fabrizio and Christian, I think the tl;dr for now is that we likely want to get rid of the polarity thing. I think @fabriziopandini will expand on it somewhere in a bit)